fastestimator/dataset/movielens20m_data.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2019 The FastEstimator Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Download Movielens-20M dataset.
+http://files.grouplens.org/datasets/movielens/ml-20m.zip
+"""
+
+import os
+import tempfile
+import zipfile
+
+import pandas as pd
+from sklearn.preprocessing import LabelEncoder, StandardScaler
+import wget
+
+GENRES = [
+    'Action', 'Adventure', 'Animation', "Children", 'Comedy', 'Crime',
+    'Documentary', 'Drama', 'Fantasy', 'Film-Noir', 'Horror', "IMAX",
+    'Musical', 'Mystery', 'Romance', 'Sci-Fi', 'Thriller', 'War', 'Western'
+]
+
+
+def _preprocess_data(path, data_path):
+    movies = pd.read_csv(os.path.join(data_path, "movies.csv"))
+    ratings = pd.read_csv(os.path.join(data_path, "ratings.csv"))
+
+    df = ratings.merge(movies, on="movieId")
+
+    for i in GENRES:
+        df[i] = df["genres"].apply(lambda x: 1 if i in x else 0)
+
+    df["genres_count"] = df["genres"].apply(lambda x: len(x.split("|")))
+    df["rating"] = df["rating"].apply(lambda x: 1 if x > 3 else 0)
+
+    df = df.drop(columns=["title", "genres"])
+
+    df["timestamp"] = pd.to_datetime(df["timestamp"], unit="s")
+    df["year"] = df["timestamp"].dt.year
+    df["month"] = df["timestamp"].dt.month
+    df["day"] = df["timestamp"].dt.day
+
+    df = df.drop(columns=["timestamp"])
+
+    LE_user = LabelEncoder()
+    df["user_id"] = LE_user.fit_transform(df["userId"])
+
+    LE_movie = LabelEncoder()
+    df["movie_id"] = LE_movie.fit_transform(df["movieId"])
+
+    # Creation of Train and Evaluation Sets
+    train_df = df.sample(frac=0.8, random_state=0)
+    eval_df = df.drop(train_df.index)
+
+    train_df = train_df.reset_index(drop=True)
+    eval_df = eval_df.reset_index(drop=True)
+
+    # Applying Standard Scaler
+    Std_cols = ["genres_count", "year", "month", "day"]
+    scaler = StandardScaler()
```

Comment:
should we be demonstrating how to do this in Pipeline instead?

Reply:
As this needs to be applied on entire dataset, we won't be able to apply this in Pipeline.